### ISSUE TITLE
extended_stimulus_presentations reward_rate calculation is incorrect

### ISSUE THREAD

**Issue by alexpiet** [AllenInstitute/visual_behavior_analysis#759]

**Comment by alexpiet** [comment]
@matchings @dougollerenshaw @yavorska-iryna This bug scares me because it calls into question other fields in extended stimulus processing! Yikes!

**Comment by matchings** [comment]
@alexpiet can you specify how you loaded the `extended_stimulus_presentations `table to see this error? did you use `loading.get_ophys_dataset()`, `then dataset.extended_stimulus_presentations`? A code block to reproduce would be helpful

**Comment by dougollerenshaw** [comment]
I agree with @matchings. Some more specifics would be helpful.

**Comment by alexpiet** [comment]
```
oeid = 1010556662
session = loading.get_ophys_dataset(oeid, include_invalid_rois=False)
esp = session.extended_stimulus_presentations
print(np.sum(esp['rewarded']) # 164
print(len(session.rewards)) #138 
```

**Comment by dougollerenshaw** [comment]
OK, I'm able to replicate the mismatch in the reward count (164 vs 138). Investigating now. I had to add imports and fix one parentheses typo to get your code block to run. Here's a full working version for reference:

```python
from visual_behavior.data_access import loading
import numpy as np

oeid = 1010556662
session = loading.get_ophys_dataset(oeid, include_invalid_rois=False)
esp = session.extended_stimulus_presentations

print(np.sum(esp['rewarded'])) # 164
print(len(session.rewards)) #138 
```

**Comment by alexpiet** [comment]
yes, sorry I dropped the imports. My head is in a different codebase right now, and I was just flagging this issue for later

**Comment by alexpiet** [comment]
How come your code block has nice colors, and mine doesnt?

**Comment by dougollerenshaw** [comment]
I just learned a new trick from @jsiegle recently. You can make github render your code with Python specific syntax highlighting by doing this:

![image](https://user-images.githubusercontent.com/19944442/119190681-88bab280-ba32-11eb-9303-c22cc2d122f5.png)

Which gives:
```python
import some_package

print(some_variable)
```

**Comment by dougollerenshaw** [comment]
Hmm, this is weird. Looking at extended stimulus presentations, I see some repeats. Look at rows 3/4 (index 60)
![image](https://user-images.githubusercontent.com/19944442/119202218-2c14c300-ba45-11eb-918d-d10bb8870ba6.png)
